### ISSUE TITLE
The openATTIC object api connection should be SSL per default

### ISSUE THREAD

**Issue by Martin-Weiss** [SUSE/DeepSea#654]
It seems that at the moment the object API connection in openATTIC is configured for clear text communication via http in case DeepSea is used.

Could this be adjusted to be SSL per default? 
(also the RGW should be SSL per default due to security reasons)

**Comment by smithfarm** [comment]
We should think carefully before having DeepSea create and deploy self-signed certificates. At the very least, they may cause browsers to display disquieting security warnings:

https://www.globalsign.com/en/ssl-information-center/dangers-self-signed-certificates/

**Comment by swiftgist** [comment]
@Martin-Weiss I do not believe we have any reliable method for making SSL the default.  Self-signed certificates cause their own support issues like smithfarm mentions and do not really address the issue.

I am not sure which set of users cannot decide that they need SSL, supply the certificate and select that configuration.  If selecting the configuration isn't obvious or simple, we should fix that.  

With respect to the Salt-API, this was originally configured as localhost.  With openATTIC as a role, the api is now listening all interfaces.  I expect that configuring the salt-api to use localhost and keeping openATTIC on the same node is secure without using SSL as well.  

As far as addressing the issue, do you think validation warnings are sufficient for those that need a secure environment?  Or do you want disabled/broken services?  I am considering a SECURE_ENV might allow us to be more aggressive.  (I don't know if one more variable is a good idea honestly.)

**Comment by Martin-Weiss** [comment]
Any self signed certificate is 100% better than having clear text communication. Standard requirement for security and compliance is to "never have any clear text authentication - especially not on the wire"...

The standard process for most applications these days (even SLES) is to create a self signed CA and create a self signed certificate signed with this CA. Then use this certificate per default for the service.

To create a proper trust chain the admin/user then needs to import the public key of the CA on the SSL client side in the trusted CA store so that no certificates warnings show up.

Many customers have their own private CA where they put the public key in the trusted CA stores of SSL clients "per default" and then they sign service-certificates with this CA so that also here the trust chain is valid.

In case external connections need to be done via SSL - a certificate is created and signed by a CA where the public key of that CA is already stored in most SSL clients trusted CA store.

So for DeepSea - I think we need these things: 

1. during default setup we should either use the servers certificate and key that exists due to the YaST CA configuration already or we should create a self signed one and use that. This allows at least to have "storage internal" communication to be encrypted and no clear text authentication will be on the wire.

2. in case we create our own self signed CA or in case we use the one from the YaST CA on the admin server - we should distribute the public key of the CA to the trusted CA store of all SSL clients within the system (openattic, rgw...)

3. we should have a central place for the certificates and keys where an admin can easily add further trusted CA certificates to be distributed (in case the certificates for rgw / openattic etc. are signed by different CAs) and where the admin can easily replace the certificates with other ones or replace then in case they are expired.

**Comment by LenzGr** [comment]
I agree with @Martin-Weiss that we should deploy oA with SSL enabled by default. I'm not familiar with the CA infrastructure that is provided by the base OS, but my expectation would be that DS uses any existing user-provided certificates and installs these on the oA node. If the user does not provide any, we should generate a self-signed cert and use that one instead.

**Comment by LenzGr** [comment]
Can't this be resolved as a duplicate of #653 ? The oA API URL is located on the same host where the oA Web UI is hosted on.
